uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChangeTest.php

```diff
@@ -109,7 +109,7 @@ public function testGetBalanceForEntitlementCanSumAllTheBalanceChangesForAGivenE
     $this->assertEquals(0.8, LeaveBalanceChange::getBalanceForEntitlement($entitlement));
   }
 
-  public function testBalanceForEntitlementCanSumOnlyTheBalanceChangesForLeaveRequestWithSpecificStatuses() {
+  public function testBalanceForEntitlementCanSumOnlyTheBalanceChangesForLeaveRequestWithSpecificStatusesAndDoesNotSumForSoftDeletedLeaveRequests() {
```

Comment:
Let's try to make tests as short as possible (I know it's a bit difficult sometimes) and avoid testing a lot of things at once. It's probably better to add a new test to only check that `balanceForEntitlement` will not include deleted leave requests.